src/wallet/rpcwallet.cpp

```diff
@@ -2262,38 +2262,107 @@ UniValue getstakesplitthreshold(const UniValue& params, bool fHelp)
     return int(pwalletMain->nStakeSplitThreshold);
 }
 
+UniValue getautocombineinfo(const UniValue& params, bool fHelp)
+{
+    if (fHelp || params.size() != 0)
+        throw runtime_error(
+            "getautocombineinfo\n"
+            "Returns the autocombinerewards settings\n"
+            "\nResult:\n"
+            "1. enabled   (string) The feature turned \"on\" or \"off\".\n"
+            "2. threshold (numeric) If enabled on, returns autocombine threshold.\n"
+            "3. frequency (variable) If enabled, returns frequency in blocks, or \"nextblock\" if one time. "
+                                    "If one time already run, \"startup\" is returned\n");
+
+    UniValue obj(UniValue::VOBJ);
+    obj.push_back(Pair("enabled", pwalletMain->fCombineDust ? "on" : "off"));
+    if (pwalletMain->fCombineDust) {
+        obj.push_back(Pair("threshold",
+                           int(pwalletMain->nAutoCombineThreshold)));
+        if (pwalletMain->nAutoCombineBlockFrequency == 0) {
+            obj.push_back(Pair("frequency", "nextblock"));
+        } else {
+            obj.push_back(Pair("frequency",
+                           int(pwalletMain->nAutoCombineBlockFrequency)));
+        }
+    }
+    else {
+        if (pwalletMain->nAutoCombineBlockFrequency == 0) {
+            obj.push_back(Pair("threshold", int(pwalletMain->nAutoCombineThreshold)));
+            obj.push_back(Pair("frequency", "startup"));
```

Comment:
See above, could use numeric value of `-2`, and add a supplemental return field with the human readable text.

Reply:
see other comments.  Changing from 0 to -2, or -1 to -2, is added complexity, where the feature is a benefit of not adding logic to make changes to the wallet database outside of the processing of autocombinerewards in rpcwallet.cpp.    The inner working is the setting of `true <#> 0` is saved, and when it executes, pwallet->fCombineDust is set to false.  That setting is not retained however, and the true remains in the database, so restarting the wallet brings it up for a single run at the start.